### PR TITLE
Refactor `virtlet` occurrences to `libvirt-provider`

### LINKED ISSUE #74: Refactoring Occurrences of `virtlet` to `libvirt-provider` and `ORI` to `IRI`
# Proposed Changes

- Rename `virtlet` to `libvirt-provider` with all it's artifacts
- Rename `ORI` to `IRI` (IronCore Runtime Interface)

### PR DESCRIPTION
**Pull request by so-sahu**
# Proposed Changes

- Rename all occurrences of `virtlet` in variable and function names to `provider`
- Rename all occurrences of `virtlet` in structure to `libvirtprovider`
- Rename all occurrences of `virtlet` in description and flags to `libvirt-provider`
- Rename all occurrences of `ori` in structure to `iri`

Fixes https://github.com/ironcore-dev/libvirt-provider/issues/74

### CODE REVIEW

**Review by lukas016 — APPROVED** [review]

File: provider/cmd/app/app.go [hunk 1]
```diff
@@ -83,8 +83,8 @@ type LibvirtOptions struct {
 }
 
 func (o *Options) AddFlags(fs *pflag.FlagSet) {
-	fs.StringVar(&o.Address, "address", "/var/run/ori-machinebroker.sock", "Address to listen on.")
-	fs.StringVar(&o.RootDir, "virtlet-dir", filepath.Join(homeDir, ".virtlet"), "Path to the directory virtlet manages its content at.")
+	fs.StringVar(&o.Address, "address", "/var/run/iri-machinebroker.sock", "Address to listen on.")
+	fs.StringVar(&o.RootDir, "libvirt-provider-dir", filepath.Join(homeDir, ".libvirt-provider"), "Path to the directory libvirt-provider manages its content at.")
```
Comment (hardikdr) [inline provider/cmd/app/app.go:87]: This will need follow-up changes in the landscape, I will handle it.

**Review by hardikdr — APPROVED** [review]
Thanks @so-sahu for the PR. Lgtm.

File: provider/cmd/app/app.go (hunk 1 above)
Comment (hardikdr) [reply provider/cmd/app/app.go:87]: cc @lukasfrank

**Comment by hardikdr** [comment]
@so-sahu you need to sign your commits  :)

File: pkg/controllers/machine_controller.go [hunk 2]
```diff
@@ -15,16 +15,16 @@ import (
...
-	virtletimage "github.com/ironcore-dev/libvirt-provider/pkg/image"
+	providerimage "github.com/ironcore-dev/libvirt-provider/pkg/image"
 	"github.com/ironcore-dev/libvirt-provider/pkg/libvirt/guest"
 	libvirtutils "github.com/ironcore-dev/libvirt-provider/pkg/libvirt/utils"
 	"github.com/ironcore-dev/libvirt-provider/pkg/os/osutils"
-	virtletnetworkinterface "github.com/ironcore-dev/libvirt-provider/pkg/plugins/networkinterface"
-	virtletvolume "github.com/ironcore-dev/libvirt-provider/pkg/plugins/volume"
+	providernetworkinterface "github.com/ironcore-dev/libvirt-provider/pkg/plugins/networkinterface"
+	providervolume "github.com/ironcore-dev/libvirt-provider/pkg/plugins/volume"
+	providerhost "github.com/ironcore-dev/libvirt-provider/pkg/providerhost" // TODO: Change to a better naming for all imports, libvirthost?
```
[the remark this answers is not in the document]
Comment (hardikdr) [reply pkg/controllers/machine_controller.go:24]: Better if the irrelevant comment is removed.

**Pull request merged**